test/test_retry.py

```diff
@@ -239,3 +239,13 @@ def test_history(self):
         self.assertEqual(retry.history, (RequestHistory('GET', '/test1', connection_error, None, None),
                                          RequestHistory('POST', '/test2', read_error, None, None),
                                          RequestHistory('GET', '/test3', None, 500, None)))
+
+    def test_retry_method_not_in_whitelist(self):
+        error = ReadTimeoutError(None, "/", "read timed out")
+        retry = Retry()
+        try:
+            retry.increment(method='POST', error=error)
+        except ReadTimeoutError:
+            pass
+        else:
+            self.fail('Request did not raise expected ReadTimeoutError')
```

Comment:
This whole block can just be replaced with `self.assertRaises(ReadTimeoutError, retry.increment, method="POST", error=error)`